### DISCUSSION TITLE
About the calculation in "tx/rx_checksum.v"

### DISCUSSION THREAD

**Discussion by linjw16** [corundum/corundum#71] [Q&A] [answered]
Hello !
  I have read the module tx_checksum & rx_checksum and checked the simulated wave. I am wondering what checksum algorithm the author is using? 

  I presume it is adding 4B (32bits)  data up to 2B (first 17 bits with overflow and then add up to 16 bits). And do this iteratively on the whole frame, i.e. accumulating all this 2B together. In that way, a register should be used to store the accumulated sum, it might be "sum_acc_reg" in line 395 in "tx_checksum.v". But the assignment is not the way I thought, is there some misunderstandings from me? 

line 395
```
else begin
            // sum_acc_reg <= sum_acc_reg+sum_acc_temp;     // linjw: might be wrong? 
            sum_acc_reg <= sum_acc_temp;
        end
```

**Comment by alexforencich** [comment, marked as answer]
See line 378

**Reply by linjw16** [reply]
Got it !
  `sum_acc_temp` is used to record the accumulated sum, while `sum_acc_reg` is used to record the last `sum_acc_temp`. 
  I had been confused between the two for a while. Thanks.

**Reply by alexforencich** [reply]
Not exactly. `sum_acc_temp` is not a register, it is a temporary variable. So the sum is being accumulated in `sum_acc_reg`, but the previous value of `sum_acc_reg` is represented in `sum_acc_temp` a few lines before `sum_acc_reg` is updated.
